.codepipeline/snapshots/snapshots.sh

```diff
@@ -4,83 +4,94 @@ set -e
+
+  aws s3 cp $sdk_name $s3_dest
 
   cd -
 }
 
-function snap_c() {
+function push_c() {
+  sdk_version=$1
+
   cd internal/wrappers/build/c
-  newname="libkuzzlesdk-c.so"
-  s3_dest="s3://$AWS_S3_BUCKET/sdk/$dest_dir$pr_name/kuzzlesdk-c.tar.gz"
-  mv libkuzzlesdk.so $newname
+
+  sdk_name="libkuzzlesdk-c.so"
+  s3_dest="s3://$AWS_S3_BUCKET/sdk/c/$sdk_version/kuzzlesdk-c.tar.gz"
+
+  mv libkuzzlesdk.so $sdk_name
   mkdir lib include
   cp ../../headers/*.h include
-  cp $newname lib/
+  cp $sdk_name lib/
   tar cfz kuzzlesdk-c.tar.gz lib include
 
-  if [[ $TRAVIS_PULL_REQUEST -ne false ]]; then
-    aws s3 cp kuzzlesdk-c.tar.gz $s3_dest --expires "$(date -d '+2 weeks' --utc +'%Y-%m-%dT%H:%M:%SZ')"
-  else
-    aws s3 cp kuzzlesdk-c.tar.gz $s3_dest
-  fi
+  aws s3 cp kuzzlesdk-c.tar.gz $s3_dest
 
   cd -
 }
 
-function snap_cpp() {
+function push_cpp() {
+  sdk_version=$1
+
   cd internal/wrappers/build/cpp
-  newname="libkuzzlesdk.so"
-  s3_dest="s3://$AWS_S3_BUCKET/sdk/$dest_dir$pr_name/kuzzlesdk-cpp.tar.gz"
-  mv libcpp.so $newname
+
+  sdk_name="libkuzzlesdk.so"
+  s3_dest="s3://$AWS_S3_BUCKET/sdk/cpp/$sdk_version/kuzzlesdk-cpp.tar.gz"
+
+  mv libcpp.so $sdk_name
   mkdir lib include
   cp ../../headers/* include
-  cp $newname lib/
+  cp $sdk_name lib/
   tar cfz kuzzlesdk-cpp.tar.gz lib include
 
-  if [[ $TRAVIS_PULL_REQUEST -ne false ]]; then
-    aws s3 cp kuzzlesdk-cpp.tar.gz $s3_dest --expires "$(date -d '+2 weeks' --utc +'%Y-%m-%dT%H:%M:%SZ')"
-  else
-    aws s3 cp kuzzlesdk-cpp.tar.gz $s3_dest
-  fi
+  aws s3 cp kuzzlesdk-cpp.tar.gz $s3_dest
 
   cd -
 }
 
+function push_sdks() {
+  sdk_version=$1
 
-if [[ -z $TRAVIS_PULL_REQUEST ]]; then
-  export dest_dir="nightly"
-else
-  pr_num="$(echo $TRAVIS_PULL_REQUEST_BRANCH | cut -d- -f2)"
-  dir_name="KZL-$pr_num"
-  export pr_name="/$dir_name"
-  export dest_dir="snapshots"
-fi
+  for dir in ${dirs[@]}; do
+    echo -e "\n----------------------------------------------------------------\n"
+    figlet "$dir SDK Snapshot"
+    echo -e "\n----------------------------------------------------------------\n"
 
-for dir in ${dirs[@]}; do
-  echo -e "\n----------------------------------------------------------------\n"
-  figlet "$dir SDK Snapshot"
-  echo -e "\n----------------------------------------------------------------\n"
 
+    push_$dir $sdk_version
 
-  snap_$dir
+    if [[ $? -ne 0 ]]; then
+      exit 1
+    fi
+  done
 
-  if [[ $? -ne 0 ]]; then
-    exit 1
+  if [[ $? -eq 0 ]]; then
+    aws cloudfront create-invalidation --distribution-id $AWS_CLOUDFRONT_DISTRIBUTION_ID --paths "/*"
   fi
-done
+}
+
+if [[ $TRAVIS_PULL_REQUEST = false ]]; then
+  echo "This script run only on push and not on PR"
+  exit 0
+fi
 
-if [[ $? -eq 0 ]]; then
-  aws cloudfront create-invalidation --distribution-id $AWS_CLOUDFRONT_DISTRIBUTION_ID --paths "/*"
+if [[ $TRAVIS_BRANCH = "master" ]]; then
+  sdk_version="latest"
+  push_sdks $sdk_version
+elif [[ $TRAVIS_BRANCH = *"-stable" ]]; then
```

Comment:
Did you check this string comparison validity? Compare strings is always tricky in Bash :stuck_out_tongue:

Reply:
Yes I tested it, you can do it too if you want: `TRAVIS_BRANCH=1-stable bash .codepipeline/snapshots/snapshots.sh`